it/uk/gov/hmrc/transitmovements/connectors/BaseUnitSpec.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovements.connectors
+
+import akka.actor.ActorSystem
+import akka.stream.Materializer
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.ExecutionContext.global
+
+trait BaseUnitSpec {
+  implicit protected val ec: ExecutionContext         = global
+  implicit protected val system: ActorSystem          = ActorSystem()
+  implicit protected val materializer: Materializer   = Materializer(system)
+  implicit protected val headerCarrier: HeaderCarrier = HeaderCarrier()
+}
```

Comment:
I'm not in principle opposed to this, but keep it in line with other repos. Find an instance of `TestActorSystem` and use that instead.

I don't think the header carrier should be provided like this in a trait. The form of it should be clear to the test because it influences the test.